src/EnvelopesQueryContext.ts

```diff
@@ -27,6 +27,11 @@ export class EnvelopesQuery {
   public getTestRunFinished(): messages.TestRunFinished | undefined {
     return this.find((envelope) => !!envelope.testRunFinished)?.testRunFinished
   }
+
+  // TODO add to cucumber query
```

Comment:
I'll raise a PR for cucumber-query. It can do this much more efficienty because it already has a map of pickle -> test case.